tests/test_simulator/immersed_body/rigid_body/test_rigid_body_forcing_grids.py

```diff
@@ -151,3 +151,173 @@ def test_circular_cylinder_grid_spacing(num_forcing_points):
     cylinder_circumference = 2 * np.pi * cylinder.radius
     correct_max_grid_spacing = cylinder_circumference / num_forcing_points
     assert correct_max_grid_spacing == max_grid_spacing
+
+
+def mock_3d_sphere():
+    """Returns a mock 3D sphere (from elastica) for testing"""
+    sphere_radius = 0.1
+    sphere_com = np.array([1.0, 2.0, 3.0])
+    sphere = ea.Sphere(center=sphere_com, base_radius=sphere_radius, density=1e3)
+    sphere.velocity_collection[...] = 3.0
+    sphere.omega_collection[...] = 4.0
+    return sphere
+
+
+@pytest.mark.parametrize("grid_dim", [2, 3])
+def test_sphere_grid_invalid_dim(grid_dim):
+    sphere = mock_3d_sphere()
+    if grid_dim != 3:
+        with pytest.raises(ValueError) as exc_info:
+            _ = sps.SphereForcingGrid(
+                grid_dim=grid_dim,
+                rigid_body=sphere,
+                num_forcing_points_along_equator=8,
+            )
+        error_msg = (
+            "Invalid grid dimensions. 3D Rigid body forcing grid is only "
+            "defined for grid_dim=3"
+        )
+        assert exc_info.value.args[0] == error_msg
+
+
+@pytest.mark.parametrize("num_forcing_points_along_equator", [8, 16])
+def test_sphere_grid_kinematics(num_forcing_points_along_equator):
+    sphere = mock_3d_sphere()
+    grid_dim = 3
+    sphere_forcing_grid = sps.SphereForcingGrid(
+        grid_dim=grid_dim,
+        rigid_body=sphere,
+        num_forcing_points_along_equator=num_forcing_points_along_equator,
+    )
+    assert sphere_forcing_grid.rigid_body is sphere
+    # number of forcing grid points is dynamic so cant test here
+    assert sphere_forcing_grid.position_field.shape[0] == grid_dim
+    assert sphere_forcing_grid.velocity_field.shape[0] == grid_dim
+    polar_angle_grid = np.linspace(0, np.pi, num_forcing_points_along_equator // 2)
+    num_forcing_points_along_latitudes = (
+        np.rint(num_forcing_points_along_equator * np.sin(polar_angle_grid)).astype(int)
+        + 1
+    )
+    assert sphere_forcing_grid.num_lag_nodes == sum(num_forcing_points_along_latitudes)
+
+    # check if grid position consistent
+    sphere_com = sphere.position_collection
+    moment_arm = sphere_forcing_grid.position_field - sphere_com
+    grid_distance_from_center = np.linalg.norm(moment_arm, axis=0)
+    # check if all points are at distance = radius from center
+    np.testing.assert_allclose(grid_distance_from_center, sphere.radius)
+    grid_centroid = np.mean(sphere_forcing_grid.position_field, axis=1)
+    np.testing.assert_allclose(grid_centroid, sphere_com[..., 0])
+
+    # check if angular locations are correct
```

Comment:
same, these tests exist in sopht already, why is the code being copy pasted as it is?